### PR TITLE
example/films: README.md and vectors tweaks

### PR DESCRIPTION
**Pull request by cpoerschke**
minor #1988 follow-up and optional support for dimensions other than 10

### CODE REVIEW

File: solr/example/films/README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 We have a movie data set in JSON, Solr XML, and CSV formats.  All 3 formats contain the same data.  You can use any one format to index documents to Solr.
 
-This example uses the `_default` configset that ships with Solr plus some custom fields added via Schema API.  It demonstrates the use of ParamSets in conjunction with the [Request Parameters API](https://solr.apache.org/guide/solr/latest/configuration-guide/request-parameters-api.html).
+This example uses the `_default` configset that ships with Solr plus some custom fields added via Schema API as described [here](https://solr.apache.org/guide/solr/latest/getting-started/tutorial-vectors.html#preparing-for-the-vector-data) in the Solr Reference Guide.  It demonstrates the use of ParamSets in conjunction with the [Request Parameters API](https://solr.apache.org/guide/solr/latest/configuration-guide/request-parameters-api.html).
```
Comment (cpoerschke) [inline solr/example/films/README.md:3]: https://solr.apache.org/guide/solr/9_5/getting-started/tutorial-vectors.html#preparing-for-the-vector-data until 'latest' becomes '9_5'

**Comment by cpoerschke** [comment]
Both the (documented) `example/films/films.json` and (undocumented) `example/films/films.xml` files work but for the (undocumented) `example/films/films.csv` file there's an error

```
Caused by: org.apache.solr.common.SolrException: incorrect vector format. The expected format is:'[f1,f2..f3]' where each element f is a float
        at org.apache.solr.util.vector.DenseVectorParser.parseIndexVector(DenseVectorParser.java:56) ~[?:?]
        at org.apache.solr.util.vector.DenseVectorParser.parseVector(DenseVectorParser.java:45) ~[?:?]
        at org.apache.solr.util.vector.FloatDenseVectorParser.getFloatVector(FloatDenseVectorParser.java:34) ~[?:?]
        at org.apache.solr.schema.DenseVectorField.createField(DenseVectorField.java:266) ~[?:?]
        at org.apache.solr.schema.DenseVectorField.createFields(DenseVectorField.java:230) ~[?:?]
        at org.apache.solr.update.DocumentBuilder.addField(DocumentBuilder.java:67) ~[?:?]
        at org.apache.solr.update.DocumentBuilder.addOriginalField(DocumentBuilder.java:304) ~[?:?]
        at org.apache.solr.update.DocumentBuilder.toDocument(DocumentBuilder.java:193) ~[?:?]
```

i.e. probably https://github.com/apache/solr/blob/releases/solr/9.4.1/solr/core/src/java/org/apache/solr/util/vector/DenseVectorParser.java#L54-L57 and perhaps it's an edge case or perhaps pipe-delimited-strings could be supported there too?

@alessandrobenedetti @epugh WDYT? I'd be happy to create a JIRA issue for that if it seems like a reasonable idea.

File: solr/example/films/README.md (hunk 1 above)
Comment (epugh) [reply solr/example/films/README.md:3]: I like!   I think in things like README's wihtout some sort of fancy versionoing tooling, you just have to piont to latest!

**Comment by epugh** [comment]
> Both the (documented) `example/films/films.json` and (undocumented) `example/films/films.xml` files work but for the (undocumented) `example/films/films.csv` file there's an error
> 
> ```
> Caused by: org.apache.solr.common.SolrException: incorrect vector format. The expected format is:'[f1,f2..f3]' where each element f is a float
>         at org.apache.solr.util.vector.DenseVectorParser.parseIndexVector(DenseVectorParser.java:56) ~[?:?]
>         at org.apache.solr.util.vector.DenseVectorParser.parseVector(DenseVectorParser.java:45) ~[?:?]
>         at org.apache.solr.util.vector.FloatDenseVectorParser.getFloatVector(FloatDenseVectorParser.java:34) ~[?:?]
>         at org.apache.solr.schema.DenseVectorField.createField(DenseVectorField.java:266) ~[?:?]
>         at org.apache.solr.schema.DenseVectorField.createFields(DenseVectorField.java:230) ~[?:?]
>         at org.apache.solr.update.DocumentBuilder.addField(DocumentBuilder.java:67) ~[?:?]
>         at org.apache.solr.update.DocumentBuilder.addOriginalField(DocumentBuilder.java:304) ~[?:?]
>         at org.apache.solr.update.DocumentBuilder.toDocument(DocumentBuilder.java:193) ~[?:?]
> ```
> 
> i.e. probably https://github.com/apache/solr/blob/releases/solr/9.4.1/solr/core/src/java/org/apache/solr/util/vector/DenseVectorParser.java#L54-L57 and perhaps it's an edge case or perhaps pipe-delimited-strings could be supported there too?
> 
> @alessandrobenedetti @epugh WDYT? I'd be happy to create a JIRA issue for that if it seems like a reasonable idea.

It is MUCH better if the CSV format works just like json and xml...   The value of the CSV format starts dropping drastically the more it doesn't support "all the things".    In fact, all of our formats should work unless there is a very clear reason why it's not possible.    

If it isn't fixable due to some constraint of the format, then yeah, we should clearly mark that in the Ref Guide.

**Comment by epugh** [comment]
Thanks for testing out the CSV format!

**Comment by epugh** [comment]
anything holding this up from being committed?

**Comment by cpoerschke** [comment]
> anything holding this up from being committed?

soft blockers only i.e.
* Solr 9.5 being released will make the link into the Solr Ref Guide work
* am still exploring more on this and might find/do further tweaks in the process

thanks for reviewing so far!

**Comment by epugh** [comment]
Nudge on this fix!

File: solr/example/films/vectors/films.py [hunk 2]
```diff
@@ -86,6 +89,7 @@ def export_films_csv(films_dataset):
         csvw = csv.DictWriter(outfile, ["name","directed_by","genre","type","id","initial_release_date","film_vector"])
         csvw.writeheader()
         for film in films_dataset:
+            # using | pipe delimiter as per TODO-add-relevant-code-reference-here
```
Comment (cpoerschke) [inline solr/example/films/vectors/films.py:92]:
```suggestion
```

not yet found (or much looked for) the relevant code reference

**Review by epugh — APPROVED** [review]
Wonderful!

**Pull request merged**